### PR TITLE
Add demo sorting by name and date (#1739)

### PR DESCRIPTION
**Pull request by ChillerDragon**
Ported from ddnet client. Demo sorting implemented by @east
https://github.com/ddnet/ddnet/pull/308

Based on work by @def-
https://github.com/ddnet/ddnet/commit/08ac551e0e6491d0795c90e0cbc46a1af0be1567

Thanks to @heinrich5991 for helping with the CDemoComparator

image

Search by name or by date ascending or descending.

### CODE REVIEW

File: src/base/system.c [hunk 1]
```diff
@@ -1647,6 +1702,15 @@ int fs_is_dir(const char *path)
 #endif
 }
 
+time_t fs_getmtime(const char *path)
+{
+	struct stat sb;
+	if(stat(path, &sb) == -1)
```
Comment (oy) [inline src/base/system.c:1708]: This probably doesn't work for older vs versions.

**Comment by ChillerDragon** [comment]
image

Nice idea with less alpha!
Not sure about the corners but yes it has to be fixed somehow.

File: src/base/system.c (hunk 1 above)
Comment (ChillerDragon) [reply src/base/system.c:1708]: Should we support older vs versions? Its in ddnet code for a while now and I haven't noticed any compile complaints about this.

**Comment by Dune-jr** [comment]
The date of the folder (the selected entry?) is gray on gray, hard to read

**Comment by ChillerDragon** [comment]
image

**Pull request merged**